### PR TITLE
Add a script to facilitate the creation of .phones files

### LINKED ISSUE #231: Improve tooling for creating .phones lists
Improve tooling for creating `.phones` lists, e.g., have a script that facilitates this task described in the [README](https://github.com/kylebgorman/wikipron/blob/fab40759d4fc95579769f44a9decf1c087771e75/data/phones/README.md): "Make a list of all phones or phonemes, in descending-frequency order".

### PR DESCRIPTION
**Pull request by jacksonllee**
This PR adds a script to facilitate the creation of the `.phones` files. I've tried to keep things really simple (both functionality and implementation), as you'll see.

Demo:

```
$ ./list_phones.py ../tsv/fre_phonemic.tsv
ʁ     # 31443     ( reculèrent | ʁ ə k y l ɛ ʁ ), ( berthe | b ɛ ʁ t ), ( surnommerions | s y ʁ n ɔ m ə ʁ j ɔ̃ )
a     # 25516     ( jaboter | ʒ a b ɔ t e ), ( pardessus | p a ʁ d ə s y ), ( xylharmonicon | ɡ z i l a ʁ m ɔ n i k ɔ̃ )
i     # 23069     ( garnissent | ɡ a ʁ n i s ), ( brandissent | b ʁ ɑ̃ d i s ), ( déchirer | d e ʃ i ʁ e )
e     # 20376     ( tutoyer | t y t w a j e ), ( réarmassions | ʁ e a ʁ m a s j ɔ̃ ), ( exerciez | ɛ ɡ z ɛ ʁ s j e )
t     # 19328     ( domestiqueront | d ɔ m ɛ s t i k ʁ ɔ̃ ), ( restockant | ʁ ə s t ɔ k ɑ̃ ), ( satané | s a t a n e )
s     # 19302     ( galiciens | ɡ a l i s j ɛ̃ ), ( maraboutassent | m a ʁ a b u t a s ), ( schizophrène | s k i z ɔ f ʁ ɛ n )
l     # 15480     ( renâclassent | ʁ ə n a k l a s ), ( hyposulfureuse | i p ɔ s y l f y ʁ ø z ), ( calomnier | k a l ɔ m n j e )
ɛ     # 14588     ( apartheid | a p a ʁ t ɛ j d ), ( épichérème | e p i k e ʁ ɛ m ), ( miellé | m j ɛ l e )
k     # 13010     ( braqueront | b ʁ a k ʁ ɔ̃ ), ( coccygienne | k ɔ k s i ʒ j ɛ n ), ( spéculateurs | s p e k y l a t œ ʁ )
ɔ     # 12426     ( eucaryotes | ø k a ʁ j ɔ t ), ( omniprésentes | ɔ m n i p ʁ e z ɑ̃ t ), ( poissonnier | p w a s ɔ n j e )
m     # 11126     ( ormeaux | ɔ ʁ m o ), ( vînmes | v ɛ̃ m ), ( aménageras | a m e n a ʒ ʁ ɑ )
j     # 11042     ( métallière | m e t a l j ɛ ʁ ), ( boycotter | b ɔ j k ɔ t e ), ( reculassiez | ʁ ə k y l a s j e )
n     # 10780     ( anoblirent | a n ɔ b l i ʁ ), ( agonisé | a ɡ ɔ n i z e ), ( aunera | o n ʁ a )
p     # 10328     ( présider | p ʁ e z i d e ), ( profilassent | p ʁ ɔ f i l a s ), ( reconceptualiseriez | ʁ ə k ɔ̃ s ɛ p t ɥ a l i z ə ʁ j e )
d     # 9965      ( sérénadé | s e ʁ e n a d e ), ( gambader | ɡ ɑ̃ b a d e ), ( mélodramatisais | m e l ɔ d ʁ a m a t i z ɛ )
ɑ̃    # 9157      ( gansait | ɡ ɑ̃ s ɛ ), ( déhanché | d e ɑ̃ ʃ e ), ( pantacourt | p ɑ̃ t a k u ʁ )
b     # 7983      ( hébertisme | e b ɛ ʁ t i s m ), ( abominé | a b ɔ m i n e ), ( bagarreur | b a ɡ a ʁ œ ʁ )
ɔ̃    # 7781      ( zaperont | z a p ʁ ɔ̃ ), ( tourbillon | t u ʁ b i j ɔ̃ ), ( accréditation | a k ʁ e d i t a s j ɔ̃ )
y     # 7201      ( bousculasses | b u s k y l a s ), ( enrubanneras | ɑ̃ ʁ y b a n ʁ ɑ ), ( prématurasses | p ʁ e m a t y ʁ a s )
z     # 6950      ( croustilleusement | k ʁ u s t i j ø z m ɑ̃ ), ( jalousions | ʒ a l u z j ɔ̃ ), ( d'occasion | d ‿ ɔ k a z j ɔ̃ )
f     # 6337      ( affirmé | a f i ʁ m e ), ( confesser | k ɔ̃ f e s e ), ( feuilleter | f œ j t e )
ə     # 5392      ( réveillonnerions | ʁ e v ɛ j ɔ n ə ʁ j ɔ̃ ), ( balafreriez | b a l a f ʁ ə ʁ j e ), ( première | p ʁ ə m j ɛ ʁ )
v     # 5362      ( improvisât | ɛ̃ p ʁ ɔ v i z ɑ ), ( verdoyer | v ɛ ʁ d w a j e ), ( avant | a v ɑ̃ )
ɡ     # 4956      ( illégalisasse | i l e ɡ a l i z a s ), ( fugues | f y ɡ ), ( gazant | ɡ a z ɑ̃ )
ʒ     # 4683      ( étageassiez | e t a ʒ a s j e ), ( prorogeassiez | p ʁ ɔ ʁ ɔ ʒ a s j e ), ( messager | m e s a ʒ e )
u     # 4234      ( antibrouillée | ɑ̃ t i b ʁ u j e ), ( goutteux | ɡ u t ø ), ( engourdir | ɑ̃ ɡ u ʁ d i ʁ )
ɑ     # 3303      ( taggâtes | t a ɡ ɑ t ), ( franchisas | f ʁ ɑ̃ ʃ i z ɑ ), ( débraseras | d e b ʁ a z ʁ ɑ )
ɛ̃    # 3274      ( interruption | ɛ̃ t ɛ ʁ y p s j ɔ̃ ), ( improviseriez | ɛ̃ p ʁ ɔ v i z ə ʁ j e ), ( insensibiliser | ɛ̃ s ɑ̃ s i b i l i z e )
o     # 3032      ( zoonose | z ɔ ɔ n o z ), ( jaunirons | ʒ o n i ʁ ɔ̃ ), ( gauchissements | ɡ o ʃ i s m ɑ̃ )
ʃ     # 2937      ( juchent | ʒ y ʃ ), ( broché | b ʁ ɔ ʃ e ), ( rafraîchirait | ʁ a f ʁ ɛ ʃ i ʁ ɛ )
w     # 2152      ( aléatoirement | a l e a t w a ʁ m ɑ̃ ), ( squatteras | s k w a t ʁ ɑ ), ( foisonneriez | f w a z ɔ n ə ʁ j e )
œ     # 1362      ( recueils | ʁ ə k œ j ), ( spammer | s p a m œ ʁ ), ( bleuisse | b l œ i s )
ø     # 1109      ( neutroalcaline | n ø t ʁ ɔ a l k a l i n ), ( ocieux | ɔ s j ø ), ( pseudonyme | p s ø d ɔ n i m )
ɥ     # 979       ( conspué | k ɔ̃ s p ɥ e ), ( buanderie | b ɥ ɑ̃ d ʁ i ), ( effectuer | e f ɛ k t ɥ e )
ɲ     # 770       ( clignais | k l i ɲ ɛ ), ( saigne | s ɛ ɲ ), ( craigne | k ʁ ɛ ɲ )
‿     # 132       ( jusqu'alors | ʒ y s k ‿ a l ɔ ʁ ), ( qu'un | k ‿ œ̃ ), ( d'époque | d ‿ e p ɔ k )
ŋ     # 86        ( woofing | w u f i ŋ ), ( surbooking | s y ʁ b u k i ŋ ), ( curling | k œ ʁ l i ŋ )
œ̃    # 65        ( juin | ʒ y œ̃ ), ( dunkerquois | d œ̃ k ɛ ʁ k w a ), ( verdun | v ɛ ʁ d œ̃ )
ɛː    # 27        ( abaissant | a b ɛː s ɑ̃ ), ( abaisseurs | a b ɛː s œ ʁ ), ( faîte | f ɛː t )
r     # 19        ( dessus | d e s y r ), ( ailleurs | a l j ɔ r ), ( crisse | k r ɪ s )
ɪ     # 15        ( icitte | i s ɪ t ), ( crisse | k r ɪ s ), ( chocolatine | ʃ ɔ k ɔ l a t s ɪ n )
h     # 13        ( honte | h ɔ̃ t ), ( hélas | h e l a ), ( haler | h a l e )
ã     # 12        ( abaissements | a b ɛː s m ã ), ( france | f ʁ ã s ), ( abaissement | a b ɛː s m ã )
æ     # 7         ( argent | æ ʁ ʒ õ ), ( bendo | b æ n d o ), ( hanout | ħ æ n uː t )
ɾ     # 6         ( frère | f ɾ æ ɾ ), ( haricot | z a ɾ i k o ), ( autres | ɔ t ɾ )
x     # 6         ( jerez | x e ʁ ɛ s ), ( manhua | m a n x w a ), ( khawef | x ä w æː f )
ɛ̯    # 5         ( ver | v a ɛ̯ ʁ ), ( mètre | m a ɛ̯ t ʁ ), ( vert | v a ɛ̯ ʁ )
ʊ     # 5         ( maudit | m o ʊ d z i ), ( tous | t ʊ s ), ( toutes | t ʊ t )
ʏ     # 4         ( lutte | l ʏ t ), ( dupe | d z ʏ p ), ( club | k l ʏ b )
d͡ʒ   # 4         ( dix | d͡ʒ i s ), ( dieu | d͡ʒ ø ), ( tadjik | t a d͡ʒ i k )
õ     # 3         ( nom | n õ ʊ̯̃ ), ( montréal | m õ ʁ e a l ), ( argent | æ ʁ ʒ õ )
ɑː    # 3         ( lézard | l e z ɑː ʁ ), ( soir | s w ɑː ʁ ), ( câlisse | k ɑː l i s )
d͡z   # 3         ( dire | d͡z i ʁ ), ( défendu | d e f ã d͡z y ), ( dix | d͡z ɪ s )
ẽ     # 3         ( matin | m a t ẽ ), ( lapin | l a p ẽ ), ( evnoniezh | ẽ w n ɔ̃ n iː ɛ s )
iː    # 3         ( jdid | d͡ʒ d iː d ), ( evnoniezh | ẽ w n ɔ̃ n iː ɛ s ), ( halloween | a l o w iː n )
ħ     # 3         ( hanout | ħ æ n uː t ), ( haouanet | ħ æ w æː n e t ), ( hass | ħ æ s )
t͡s   # 3         ( tuer | t͡s ɥ e ), ( tuer | t͡s y e ), ( petite | p ə t͡s ɪ t )
ɒ     # 2         ( arrondissement | ə ɹ ɒ n d ɪ s m ə n t ), ( choc | t ʃ ɒ k )
ɔː    # 2         ( soir | s w ɔː ʁ ), ( fjord | f j ɔː ʁ )
ʊː    # 2         ( gouges | ɡ ʊː ʒ ), ( gougent | ɡ ʊː ʒ )
ʔ     # 2         ( héhé | e ʔ e ), ( haha | a ʔ a )
ɚ     # 2         ( loser | l u z ɚ ), ( hamster | a m s t ɚ )
æː    # 2         ( khawef | x ä w æː f ), ( haouanet | ħ æ w æː n e t )
ä     # 2         ( tarin | t ä ʁ æ̃ ), ( khawef | x ä w æː f )
ɹ     # 1         ( arrondissement | ə ɹ ɒ n d ɪ s m ə n t )
kʲ    # 1         ( chèque | ʃ ɛ kʲ )
ʌ     # 1         ( fucker | f ʌ k e )
ɜ     # 1         ( gougèrent | ɡ u ʒ ɜ ʁ )
uː    # 1         ( hanout | ħ æ n uː t )
œː    # 1         ( heure | œː ʁ )
ɶ̝    # 1         ( honneur | ɔ n ɶ̝ ʁ )
lː    # 1         ( illustré | i lː y s t ʁ e )
ɳ     # 1         ( imprégnation | ɛ̃ p ʁ e ɳ a s j ɔ̃ )
ɛ̃ː   # 1         ( même | m ɛ̃ː m )
ʊ̯̃   # 1         ( nom | n õ ʊ̯̃ )
p̚    # 1         ( ouaip | w ɛ p̚ )
ʎ     # 1         ( spremiaglio | s p r e m i a ʎ ʎ o )
ĩᵑ    # 1         ( tarin | t a ʁ ĩᵑ )
æ̃    # 1         ( tarin | t ä ʁ æ̃ )
t͡ʃ   # 1         ( tuer | t͡ʃ ɥ e )
˞     # 1         ( un | œ̃ ˞ )

# unique phones: 81
```

Closes #231.

- [x] Updated `Unreleased` in `CHANGELOG.md` to reflect the changes in code or data.

### CODE REVIEW

File: data/phones/list_phones.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+def _get_cli_args() -> argparse.Namespace:
+    parser = argparse.ArgumentParser(description=__doc__)
+    parser.add_argument("filepath", help="Path to TSV scraped by WikiPron")
+    args = parser.parse_args()
+    return args
+
+
+def _count_phones(filepath: str) -> Dict[str, Set[str]]:
+    # phone_to_examples as Dict[str, Set[str]] is the most straightforward
```
Comment (kylebgorman) [inline data/phones/list_phones.py:26]: What about changing this to a [PEP 257](https://www.python.org/dev/peps/pep-0257/) docstring instead?

File: data/phones/list_phones.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+            word, pron = line.split("\t", maxsplit=1)
+            example = f"( {word} | {pron} )"
+            phones = pron.split()
+            for phone in phones:
+                phone_to_examples[phone].add(example)
+    return phone_to_examples
+
+
+def _pick_examples_for_display(examples: Set[str]) -> List[str]:
+    # We could have exposed the maximum number of examples to display
```
Comment (kylebgorman) [inline data/phones/list_phones.py:45]: Ditto?

File: data/phones/list_phones.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    # (set to be 3 now) to the command-line interface,
+    # but it doesn't seem worth it for the time being.
+    n_examples = min(len(examples), 3)
+    # Using list() here because Python 3.9 has deprecated the use
+    # of an _unordered_ set as the input to random.sample.
+    sample = random.sample(list(examples), n_examples)
```
Comment (kylebgorman) [inline data/phones/list_phones.py:51]: I would prefer `return random.sample(list(examples), n_examples)`; not seeing any reason to save to named variable as an intermediate step. (Calling it `sample` doesn't really add any information IMO.)

File: data/phones/list_phones.py [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    ):
+        print(
+            f"{phone:<5}"
+            f"{len(examples):<10}"
+            f"{', '.join(_pick_examples_for_display(examples))}"
+        )
+    print(f"\n# unique phones: {len(phone_to_examples)}")
+
+
+if __name__ == "__main__":
```
Comment (kylebgorman) [inline data/phones/list_phones.py:69]: I sort of prefer the style where argparsing goes here and is passed to main (I also don't think it has any good reason to be its own function). I believe it's what we've used with other tools, or it was what I used initially.

As is `main` is not reusable except in a command-line environment, which is the purpose of having `main` be its own function, is it not?

**Comment by kylebgorman** [comment]
Hi Jackson, thanks for this. I added some finicky suggestions for style, do with them what you will.

Could you possibly add some documentation for how this ought to be used to the directory's README?

**Comment by jacksonllee** [comment]
Hey Kyle, it looks like your comments didn't come through on GitHub?

Re: documentation, I've already updated `data/phones/README.md`. Let me know if you'd like me to adjust anything.

File: data/phones/README.md [hunk 5]
```diff
@@ -36,7 +36,13 @@ We welcome user submissions for `.phones` files from linguists. Note that we use
 
 1.  Make a list of all phones or phonemes, in descending-frequency order, using
     the appropriate file in [`../tsv`](../tsv).
+    The script `list_phones.py` is available to facilitate this step.
+    Running `./list_phones.py ../tsv/<some-TSV-file> > foo.phones`
+    generates `foo.phones` that you can edit by the following steps.
 2.  Remove typos, invalid IPA transcriptions, and non-native segments.
+    The `.phones` file generated by `list_phones.py` shows the frequency of each
+    phone/phoneme and several of its example word-pronunciation pairs,
+    which should help you decide which phones/phonemes to remove.
```
Comment (kylebgorman) [inline data/phones/README.md:45]: Maybe remind them to remove comments and counts.

It occurs to me that maybe to be sure, you could comment out the counts too?

**Comment by jacksonllee** [comment]
> It occurs to me that maybe to be sure, you could comment out the counts too?

If I see what you mean, I think it's a good idea just in case folks didn't remember to remove the counts and example word-pron pairs (which has been added as a reminder in the README). I [updated the comment above](https://github.com/kylebgorman/wikipron/pull/246#issue-507237479) to show `#` after the phone/phoneme on each line.

Let me know if you want me to adjust anything else.

**Review by kylebgorman — APPROVED** [review]
LGTM

**Pull request merged**